src/Mapster/Adapters/ClassAdapter.cs

```diff
@@ -23,10 +23,11 @@ protected override bool CanMap(Type sourceType, Type destinationType, MapType ma
             if (sourceType == typeof(string) || sourceType == typeof(object))
                 return false;
 
-            if (!destinationType.IsPoco())
+            if (destinationType.GetTypeInfo().IsEnum || destinationType.Namespace.StartsWith("System"))
                 return false;
 
-            return true;
+            return (destinationType.GetPublicFieldsAndProperties(allowNoSetter: false).Any() ||
+                    destinationType.GetPrivateFieldsAndProperties(allowNoSetter: false).Any());
```

Comment:
I added ClassWithNonPublicMemberAdapter and reverted back the ClassAdapter. For now the score for both classes are the same. Please let me know if further changes needed.